src/app/data/oasis-account-names.ts

```diff
@@ -0,0 +1,110 @@
+import axios from 'axios'
+import { useQuery } from '@tanstack/react-query'
+import { Layer } from '../../oasis-nexus/api'
+import { Network } from '../../types/network'
+import { hasTextMatch } from '../components/HighlightedText/text-matching'
+import { AccountData, AccountEntry, AccountMap, AccountMetadata, AccountMetadataInfo } from './named-accounts'
+
+const dataSources: Record<Network, Partial<Record<Layer, string>>> = {
+  [Network.mainnet]: {
+    [Layer.consensus]:
+      'https://raw.githubusercontent.com/oasisprotocol/nexus/main/account-names/mainnet_consensus.json',
+    [Layer.emerald]:
+      'https://raw.githubusercontent.com/oasisprotocol/nexus/main/account-names/mainnet_paratime.json',
+    [Layer.sapphire]:
+      'https://raw.githubusercontent.com/oasisprotocol/nexus/main/account-names/mainnet_paratime.json',
+  },
+  [Network.testnet]: {
+    [Layer.consensus]:
+      'https://raw.githubusercontent.com/oasisprotocol/nexus/main/account-names/testnet_consensus.json',
+    [Layer.emerald]:
+      'https://raw.githubusercontent.com/oasisprotocol/nexus/main/account-names/testnet_paratime.json',
+    [Layer.sapphire]:
+      'https://raw.githubusercontent.com/oasisprotocol/nexus/main/account-names/testnet_paratime.json',
+  },
+}
+
+const getOasisAccountsMetadata = (network: Network, layer: Layer) =>
+  new Promise<AccountData>((resolve, reject) => {
+    const url = dataSources[network][layer]
+    if (!url) {
+      reject('No data available for this layer')
+    } else {
+      axios.get(url).then(response => {
+        if (response.status !== 200) reject("Couldn't load names")
+        if (!response.data) reject("Couldn't load names")
+        // console.log('Response data is', response.data)
+        const map: AccountMap = new Map()
+        const list: AccountEntry[] = []
+        Array.from(response.data).forEach((entry: any) => {
+          const address = entry.Address
+          const metadata: AccountMetadata = {
+            name: entry.Name,
+            description: entry.Description,
+          }
+          // Register the metadata in its native form
+          list.push({
+            address,
+            ...metadata,
+          })
+          map.set(address, metadata)
+        })
+        resolve({
+          map,
+          list,
+        })
+      }, reject)
+    }
+  })
+
+export const useOasisAccountsMetadata = (network: Network, layer: Layer, enabled: boolean) => {
+  return useQuery(['oasisAccounts', network, layer], () => getOasisAccountsMetadata(network, layer), {
+    enabled,
+    staleTime: Infinity,
+  })
+}
+
+export const useOasisAccountMetadata = (
+  network: Network,
+  layer: Layer,
+  address: string,
+  enabled: boolean,
+): AccountMetadataInfo => {
+  // This is not a condition that can change while the app is running, so it's OK.
+  // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
Yeah, code comment was obsolete, removed. (There was no condition here whatsoever, so the problem the comment has originally been described has since been solved a different way.)